internal/api/keppel/accounts.go

```diff
@@ -542,6 +544,66 @@ func (a *API) handlePutAccount(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 
+		// Copy PlatformFilter when creating an account with the Replication Policy on_first_use
+		if req.Account.ReplicationPolicy != nil {
+			rp := *req.Account.ReplicationPolicy
+			if rp.Strategy == "on_first_use" {
+				var peer keppel.Peer
+				err := a.db.SelectOne(&peer, `SELECT * FROM peers WHERE hostname = $1`, rp.UpstreamPeerHostName)
+				if err == sql.ErrNoRows {
+					http.Error(w, fmt.Sprintf(`unknown peer registry: %q`, rp.UpstreamPeerHostName), http.StatusUnprocessableEntity)
+					return
+				}
+				if respondwith.ErrorText(w, err) {
+					return
+				}
+
+				reqURL := fmt.Sprintf("https://%s/peer/v1/account-filter/%s", accountToCreate.UpstreamPeerHostName, accountToCreate.Name)
+				authReq, err := http.NewRequest("GET", reqURL, nil)
+				if respondwith.ErrorText(w, err) {
+					return
+				}
+				//TODO: a.cfg.OurReplicationUsername = fmt.Sprintf("replication@%s", a.cfg.APIPublicURL.Hostname())
+				authReq.Header.Set("Authorization", keppel.BuildBasicAuthHeader(fmt.Sprintf("replication@%s", a.cfg.APIPublicURL.Hostname()), peer.OurPassword))
+
+				resp, err := http.DefaultClient.Do(authReq)
+				if err != nil {
+					http.Error(w, "could not validate credentials: "+err.Error(), http.StatusUnauthorized)
+					return
+				}
+				if resp.StatusCode != http.StatusOK {
+					http.Error(w, "could not validate credentials: expected 200 OK, but got "+resp.Status, http.StatusUnauthorized)
+					return
+				}
+
+				respBodyBytes, err := ioutil.ReadAll(resp.Body)
+				if err == nil {
+					err = resp.Body.Close()
+				} else {
+					resp.Body.Close()
+				}
+				if err != nil {
+					return
+				}
+
+				var data peerv1.AccountFilter
+				err = json.Unmarshal(respBodyBytes, &data)
+				if err != nil {
+					return
+				}
+
+				if req.Account.PlatformFilter == nil {
+					accountToCreate.PlatformFilter = data.Filter
+				} else if !reflect.DeepEqual(req.Account.PlatformFilter, data.Filter) {
+					// check if the peer PlatformFilter matches the primary account PlatformFilter
+					JSONPlatformFilter, _ := json.Marshal(req.Account.PlatformFilter)
+					JSONFilter, _ := json.Marshal(data.Filter)
+					http.Error(w, fmt.Sprintf("peer account filter needs to match primary account filter: primary account %s, peer account %s ", JSONPlatformFilter, JSONFilter), http.StatusConflict)
```

Comment:
Identifiers for locally-scoped variables should start in lower case by convention. (I'm actually surprised that staticcheck does not flag this.)
```suggestion
					jsonPlatformFilter, _ := json.Marshal(req.Account.PlatformFilter)
					jsonFilter, _ := json.Marshal(data.Filter)
					http.Error(w, fmt.Sprintf("peer account filter needs to match primary account filter: primary account %s, peer account %s ", jsonPlatformFilter, jsonFilter), http.StatusConflict)
```

Also, line 601 is quite long and could be broken up a bit. When messages get very long, I usually send the sprintf into a temp variable first, like so:
```go
msg := fmt.Sprintf("peer account filter needs to match primary account filter: primary account %s, peer account %s ",
  jsonPlatformFilter, jsonFilter)
http.Error(w, msg, http.StatusConflict)
```